corporate/lib/stripe.py

```diff
@@ -1049,6 +1058,109 @@ def downgrade_small_realms_behind_on_payments_as_needed() -> None:
                 void_all_open_invoices(realm)
 
 
+def switch_realm_from_standard_to_plus(realm: Realm) -> None:
+    standard_plan = get_current_plan_by_realm(realm)
+
+    if not standard_plan:
+        raise BillingError("Organization does not have any active standard_plan")
+
+    if not standard_plan.customer.stripe_customer_id:
+        raise BillingError("Organization missing stripe customer ID")
+
+    current_datetime = datetime.now()
+
+    # Send invoice for additional users added since the billing period started till now.
+    invoice_plan(standard_plan, datetime.now())
+
+    last_renewal_ledger = LicenseLedger.objects.filter(
+        is_renewal=True, plan=standard_plan
+    ).order_by("id")
+    last_ledger = LicenseLedger.objects.filter(plan=standard_plan).order_by("id")
+
+    last_renewal_amount = last_renewal_ledger.licenses * standard_plan.price_per_license
+
+    last_renewal_date = last_renewal_ledger.event_time
+    next_renewal_date = start_of_next_billing_cycle(standard_plan, current_datetime)
+
+    unused_period_fraction = (current_datetime - last_renewal_date) / (
+        next_renewal_date - last_renewal_date
+    )
+    amount_to_credit_back_to_realm = last_renewal_amount * unused_period_fraction
+
+    stripe.Customer.create_balance_transaction(
+        standard_plan.customer.stripe_customer_id,
+        amount=-1 * amount_to_credit_back_to_realm,
+        currency="usd",
+        description="Standard plan credit",
+    )
+
+    standard_plan.status = CustomerPlan.ENDED
+    standard_plan.save(update_fields=["status"])
+
+    discount = standard_plan.customer.default_discount or standard_plan.discount
+    (
+        billing_cycle_anchor,
+        next_invoice_date,
+        period_end,
+        price_per_license,
+    ) = compute_plan_parameters(
+        automanage_licenses=standard_plan.automanage_licenses,
+        billing_schedule=standard_plan.billing_schedule,
+        discount=discount,
+    )
+
+    plus_plan = CustomerPlan.objects.create(
+        automanage_licenses=standard_plan.automanage_licenses,
+        charge_automatically=standard_plan.charge_automatically,
+        price_per_license=price_per_license,
+        discount=discount,
+        billing_schedule=standard_plan.billing_schedule,
+        tier=CustomerPlan.PLUS,
```

Comment:
Is the rest of our code equipped to handle the PLUS value? E.g. will it correctly show up at /activity/support page of the organization, or what will org admins see at their /billing/ as the active plan?

Reply:
Yeah. I think those two pages make use of plan.name so should work correctly.

I think one change that we should probably make is to add `PLUS` as a plan_type to Realm. It's not something that I have not made in this PR.  Maybe we should do that as well.